cypress/e2e/signIn.cy.js

```
@@ -2,10 +2,51 @@
 
 describe('Sign In page', () => {
   beforeEach(() => {
+    cy.visit('login');
+  });
+
+  it('should allow log in with valid creds', () => {
+    cy.get('#username')
+      .type('tomsmith');
+
+    cy.get('#password')
+      .type('SuperSecretPassword!');
+
+    cy.get('.fa')
+      .click();
+
+    cy.get('#flash')
+      .should('contain.text', 'You logged into a secure area!');
+  });
 
+  it('should not allow log in with invalid creds', () => {
+    cy.get('#username')
+      .type('user123');
+
+    cy.get('#password')
+      .type('passworduser123');
+
+    cy.get('.fa')
+      .click();
+
+    cy.get('#flash')
+      .should('contain.text', 'Your username is invalid!');
   });
 
-  it('', () => {
+  it('should allow log out from the app', () => {
+    cy.get('#username')
+      .type('tomsmith');
+
+    cy.get('#password')
+      .type('SuperSecretPassword!');
+
+    cy.get('.fa')
+      .click();
```

Logout test: before clicking the logout control you should assert that the login succeeded (for example `cy.url().should('include', '/secure')` or check logout button visibility). This makes the test explicit about the precondition for logout and prevents false positives if login failed silently.